dbt/adapters/redshift/connections.py

```diff
@@ -39,13 +40,16 @@ def json_schema(self):
 class RedshiftConnectionMethod(StrEnum):
     DATABASE = "database"
     IAM = "iam"
+    AUTH_PROFILE = "auth_profile"
+    IDP = "IdP"
 
 
 @dataclass
 class RedshiftCredentials(Credentials):
-    host: str
-    user: str
     port: Port
+    region: Optional[str] = None
```

Comment:
Can we infer from the host value the region here like `host.split(".")[2]`?

Reply:
since host isn't required anymore(auth_profile doesn't require host) setting region = host.split(".")[2] will fail. What I've done is setting region to host.split(".")[2] in the get_connec_method() call if the connection method is iam or database

Reply:
makes sense!